src/rez/utils/request_directives.py

```diff
@@ -0,0 +1,280 @@
+
```

Comment:
The functions and classes here really require more explanation, with examples. It is quite hard to follow what's going on.

Reply:
Yeah, sorry about that 😅
doc string coming !